### DISCUSSION TITLE
可否在readme加入如何支持第三方DDNS

### DISCUSSION THREAD

**Discussion by woshichenghaibo** [jeessy2/ddns-go#1060] [General]
很多运营商其实是有自己的API，可以实现DDNS的，比如我想实现netlify的DDNS，现在可以用bash脚本实现，代码在：

https://github.com/woshichenghaibo/Netlify-DDNS/blob/main/v6ddns.sh

不知道怎么才能改为使用咱们ddnsgo实现呢？

谢谢

**Comment by jeessy2** [comment]
写代码实现。
或直接用`callback`调用接口，未测试

**Comment by woshichenghaibo** [comment]
> 写代码实现。 或直接用`callback`调用接口，未测试

已经用bash代码实现了，但是还是想用ddnsgo实现，就是不知道具体怎么操作，可否写在readme里面

**Reply by jeessy2** [reply]
意思是加链接？这个不行吧。都不是一套

**Reply by woshichenghaibo** [reply]
我的意思是说明下怎么使用callback，或者ddnsgo有无配置文件可以使用模板自己添加服务商

**Reply by jeessy2** [reply]
callback你参考文档填写测试下就可以了。没有模板自己添加服务商，要添加单独的，就得写代码

**Reply by woshichenghaibo** [reply]
ddnsgo自定义服务商 回调方法没看懂，可能你是为高手准备的。

回调的url写在哪里了？点哪个入口进去？还是随便选择上面的服务商，在ipv6的域名那个空格里写自定义服务商？

**Reply by jeessy2** [reply]
参考 https://github.com/jeessy2/ddns-go/wiki/Callback%E9%85%8D%E7%BD%AE%E5%8F%82%E8%80%83

你都会写脚本了，这个对你小问题一个

**Reply by woshichenghaibo** [reply]
是 :/etc/ddns-go/ddnsgo-config.yaml 这里面写回调吗？

**Reply by jeessy2** [reply]
网页上，服务商不是有一个callback?

**Reply by woshichenghaibo** [reply]
我有脚本但是不知道写在什么地方，你的readme没写在哪里填写

**Reply by woshichenghaibo** [reply]
啊，看到了，嗨，你把callback跟其他放在一起，既没放最前面也没放最后面，还以为都是正常厂家名字呢

**Reply by jeessy2** [reply]
不能调脚本，二套不同的，ddns-go和你脚本无关系

**Reply by woshichenghaibo** [reply]
建议这个选项放最前或者最后或者标注个红色。。。

**Reply by jeessy2** [reply]
根据开发顺序来的，这个不用调顺序

**Reply by woshichenghaibo** [reply]
请教，我刚安装完还没添加，我的日志显示 这些，是不是要修改哪里才行？

2024/03/28 03:04:06 等待网络连接: lookup domains.google.com on [::1]:53: read udp 192.168.3.151:40541->1.1.1.1:53: i/o timeout
2024/03/28 03:04:06 5s 后重试...
2024/03/28 03:04:51 等待网络连接: lookup dns.myhuaweicloud.com on [::1]:53: read udp 192.168.3.151:49106->1.1.1.1:53: i/o timeout
2024/03/28 03:04:51 5s 后重试...
2024/03/28 03:05:36 等待网络连接: lookup dynamicdns.park-your-domain.com on [::1]:53: read udp 192.168.3.151:49855->1.1.1.1:53: i/o timeout
2024/03/28 03:05:36 5s 后重试...
2024/03/28 03:06:22 等待网络连接: lookup www.namesilo.com on [::1]:53: read udp 192.168.3.151:45814->1.1.1.1:53: i/o timeout
2024/03/28 03:06:22 5s 后重试...
2024/03/28 03:07:07 等待网络连接: lookup porkbun.com on [::1]:53: read udp 192.168.3.151:33346->1.1.1.1:53: i/o timeout
2024/03/28 03:07:07 5s 后重试...
2024/03/28 03:07:52 等待网络连接: lookup dnspod.tencentcloudapi.com on [::1]:53: read udp 192.168.3.151:53551->1.1.1.1:53: i/o timeout
2024/03/28 03:07:52 5s 后重试...
2024/03/28 03:08:37 等待网络连接: lookup alidns.aliyuncs.com on [::1]:53: read udp 192.168.3.151:53572->1.1.1.1:53: i/o timeout
2024/03/28 03:08:37 5s 后重试...
2024/03/28 03:09:22 等待网络连接: lookup bcd.baidubce.com on [::1]:53: read udp 192.168.3.151:39590->1.1.1.1:53: i/o timeout
2024/03/28 03:09:22 5s 后重试...
2024/03/28 03:10:07 等待网络连接: lookup api.cloudflare.com on [::1]:53: read udp 192.168.3.151:34901->1.1.1.1:53: i/o timeout
2024/03/28 03:10:07 5s 后重试...
2024/03/28 03:10:52 等待网络连接: lookup dnsapi.cn on [::1]:53: read udp 192.168.3.151:35326->1.1.1.1:53: i/o timeout
2024/03/28 03:10:52 5s 后重试...
2024/03/28 03:11:37 等待网络连接: lookup domains.google.com on [::1]:53: read udp 192.168.3.151:55026->1.1.1.1:53: i/o timeout
2024/03/28 03:11:37 5s 后重试...
2024/03/28 03:12:22 等待网络连接: lookup dns.myhuaweicloud.com on [::1]:53: read udp 192.168.3.151:55214->1.1.1.1:53: i/o timeout
2024/03/28 03:12:22 5s 后重试...
2024/03/28 03:13:07 等待网络连接: lookup dynamicdns.park-your-domain.com on [::1]:53: read udp 192.168.3.151:57229->1.1.1.1:53: i/o timeout
2024/03/28 03:13:07 5s 后重试...
2024/03/28 03:13:52 等待网络连接: lookup www.namesilo.com on [::1]:53: read udp 192.168.3.151:59724->1.1.1.1:53: i/o timeout
2024/03/28 03:13:52 5s 后重试...
2024/03/28 03:14:37 等待网络连接: lookup porkbun.com on [::1]:53: read udp 192.168.3.151:52177->1.1.1.1:53: i/o timeout
2024/03/28 03:14:37 5s 后重试...
2024/03/28 03:15:22 等待网络连接: lookup dnspod.tencentcloudapi.com on [::1]:53: read udp 192.168.3.151:56291->1.1.1.1:53: i/o timeout
2024/03/28 03:15:22 5s 后重试...
2024/03/28 03:16:07 等待网络连接: lookup alidns.aliyuncs.com on [::1]:53: read udp 192.168.3.151:41637->1.1.1.1:53: i/o timeout
2024/03/28 03:16:07 5s 后重试...
2024/03/28 03:16:52 等待网络连接: lookup bcd.baidubce.com on [::1]:53: read udp 192.168.3.151:58112->1.1.1.1:53: i/o timeout
2024/03/28 03:16:52 5s 后重试...
2024/03/28 03:17:37 等待网络连接: lookup api.cloudflare.com on [::1]:53: read udp 192.168.3.151:37656->1.1.1.1:53: i/o timeout
2024/03/28 03:17:37 5s 后重试...
2024/03/28 03:18:22 等待网络连接: lookup dnsapi.cn on [::1]:53: read udp 192.168.3.151:37049->1.1.1.1:53: i/o timeout
2024/03/28 03:18:22 5s 后重试...
2024/03/28 03:19:07 等待网络连接: lookup domains.google.com on [::1]:53: read udp 192.168.3.151:34300->1.1.1.1:53: i/o timeout
2024/03/28 03:19:07 5s 后重试...
2024/03/28 03:19:52 等待网络连接: lookup dns.myhuaweicloud.com on [::1]:53: read udp 192.168.3.151:52500->1.1.1.1:53: i/o timeout
2024/03/28 03:19:52 5s 后重试...
2024/03/28 03:20:37 等待网络连接: lookup dynamicdns.park-your-domain.com on [::1]:53: read udp 192.168.3.151:36121->1.1.1.1:53: i/o timeout
2024/03/28 03:20:37 5s 后重试...
2024/03/28 03:21:22 等待网络连接: lookup www.namesilo.com on [::1]:53: read udp 192.168.3.151:35721->1.1.1.1:53: i/o timeout
2024/03/28 03:21:22 5s 后重试...
2024/03/28 03:22:07 等待网络连接: lookup porkbun.com on [::1]:53: read udp 192.168.3.151:39052->1.1.1.1:53: i/o timeout
2024/03/28 03:22:07 5s 后重试...

**Reply by jeessy2** [reply]
openwrt?要升级最新，好像插件未发布

**Reply by woshichenghaibo** [reply]
是我的dns设置的问题吗

**Reply by jeessy2** [reply]
不是，算bug吧。你下6.3.0安装包运行吧

**Comment by BiFangKNT** [comment]
@jeessy2 已经一年半了，希望官方支持netlify

**Comment by jeessy2** [comment]
> @jeessy2 已经一年半了，希望官方支持netlify

可以PR
